src/RequestMatcher.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Http\Client\Common;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * RequestMatcher allow to tell if a PSR7 Request matches a specific strategy
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+interface RequestMatcher
```

Comment:
We did, message. See https://github.com/php-http/client-common/pull/14#issuecomment-203115143